File: wealth-dashboard/src/data/sampleData.js
const generatePerformanceData = () => {
  const dates = [];
  const values = [];
  const initialValue = 400000;
  
  for (let i = 0; i < 365; i++) {
    const date = new Date();
    date.setDate(date.getDate() - (365 - i));
    dates.push(date.toISOString());
    
    // Generate smoother value progression
    const trend = i / 365 * 0.15; // 15% annual growth trend
    const variation = (Math.random() - 0.5) * 0.02; // Small random variations
    values.push(initialValue * (1 + trend + variation));
  }
  
  return { dates, values };
};

export const portfolioData = {
  holdings: [
    {
      asset: 'US Stocks',
      currentValue: 250000,
      purchasePrice: 200000,
      gainLoss: 50000,
      weight: 50,
      return: 25
    },
    {
      asset: 'International Stocks',
      currentValue: 100000,
      purchasePrice: 90000,
      gainLoss: 10000,
      weight: 20,
      return: 11.11
    },
    {
      asset: 'Bonds',
      currentValue: 75000,
      purchasePrice: 72000,
      gainLoss: 3000,
      weight: 15,
      return: 4.17
    },
    {
      asset: 'Real Estate',
      currentValue: 50000,
      purchasePrice: 45000,
      gainLoss: 5000,
      weight: 10,
      return: 11.11
    },
    {
      asset: 'Cash',
      currentValue: 25000,
      purchasePrice: 25000,
      gainLoss: 0,
      weight: 5,
      return: 0
    }
  ],
  allocation: {
    labels: ['US Stocks', 'International Stocks', 'Bonds', 'Real Estate', 'Cash'],
    values: [50, 20, 15, 10, 5]
  },
  performance: generatePerformanceData()
};

export const calculateMetrics = (data) => {
  const totalValue = data.holdings.reduce((sum, holding) => sum + holding.currentValue, 0);
  const totalCost = data.holdings.reduce((sum, holding) => sum + holding.purchasePrice, 0);
  const performance = data.performance;
  
  return {
    totalValue,
    totalGain: totalValue - totalCost,
    dailyChange: ((performance.values[performance.values.length - 1] - 
                  performance.values[performance.values.length - 2]) / 
                  performance.values[performance.values.length - 2] * 100),
    gainPercentage: ((totalValue - totalCost) / totalCost * 100),
    roi: ((totalValue - totalCost) / totalCost * 100),
    roiChange: ((performance.values[performance.values.length - 1] - performance.values[0]) / 
                performance.values[0] * 100)
  };
};

File: data/sample_data.py
import pandas as pd
import numpy as np
from datetime import datetime, timedelta

def get_portfolio_data():
    """Generate sample portfolio data"""
    # Generate sample holdings data
    holdings = [
        {
            'Asset': 'US Stocks',
            'Current Value': 250000,
            'Purchase Price': 200000,
            'Gain/Loss': 50000,
            'Weight': 50,
            'Return': 25
        },
        {
            'Asset': 'International Stocks',
            'Current Value': 100000,
            'Purchase Price': 90000,
            'Gain/Loss': 10000,
            'Weight': 20,
            'Return': 11.11
        },
        {
            'Asset': 'Bonds',
            'Current Value': 75000,
            'Purchase Price': 72000,
            'Gain/Loss': 3000,
            'Weight': 15,
            'Return': 4.17
        },
        {
            'Asset': 'Real Estate',
            'Current Value': 50000,
            'Purchase Price': 45000,
            'Gain/Loss': 5000,
            'Weight': 10,
            'Return': 11.11
        },
        {
            'Asset': 'Cash',
            'Current Value': 25000,
            'Purchase Price': 25000,
            'Gain/Loss': 0,
            'Weight': 5,
            'Return': 0
        }
    ]
    
    # Generate sample allocation data
    allocation = {
        'labels': ['US Stocks', 'International Stocks', 'Bonds', 'Real Estate', 'Cash'],
        'values': [50, 20, 15, 10, 5]
    }
    
    # Generate sample performance data
    dates = pd.date_range(end=datetime.now(), periods=365).tolist()
    initial_value = 400000
    np.random.seed(42)
    random_walk = np.random.normal(loc=0.0002, scale=0.001, size=365).cumsum()
    values = (initial_value * (1 + random_walk)).tolist()
    
    performance = {
        'dates': dates,
        'values': values
    }
    
    return {
        'holdings': holdings,
        'allocation': allocation,
        'performance': performance
    }
